src/priorityqueue.jl

```diff
@@ -351,13 +357,71 @@ function empty!(pq::PriorityQueue)
     pq
 end
 
+empty(pq::PriorityQueue) = PriorityQueue(empty(pq.xs), pq.o, empty(pq.index))
+
+function merge!(d::SortedDict{K, V, O1}, other::PriorityQueue{K, V, O}) where {K, V, O <: Ordering, O1 <: Ordering}
+    next = iterate(other, false)
+    while next !== nothing
+        (k, v), state = next
+        d[k] = v
+        next = iterate(other, state)
+    end
+    return d
+end
+
+function merge!(d::AbstractDict{K, V}, other::PriorityQueue{K, V, O}) where {K, V, O <: Ordering}
```

Comment:
You can drop the type parameters here and below.  In Julia, it's common not to enforce that containers have the same key/element types.  If the types in the second collection are not convertible to the first, an error will be thrown.